enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,203 @@
+}
+
+type AutoDetectVolumeStatus struct {
+	Status string `json:"status"`
+}
+
+type DeviceDiscoveryPolicy struct {
+	// Device type that should be used for auto detection. This would be one of the types supported
+	// by the local-storage operator. Initial possible types can be - crypt, raid1, raid4, raid5,
+	// raid10, multipath, disk, tape, printer, processor, worm, rom, scanner, mo-disk, changer,
+	// comm, raid, enclosure, rbc, osd, and no-lun.
+	DeviceType DeviceDiscoveryPolicyType `json:"deviceType"`
+	// File system type
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// Volume mode. Raw or with file system
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// DeviceInclusionSpec are the filtration rules for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational,nonRotational and
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	Rotational                 DeviceMechanicalProperty = "rotational"
+	NonRotational              DeviceMechanicalProperty = "nonRotational"
+	RotationalAndNonRotational DeviceMechanicalProperty = "rotationalAndNonRotational"
+)
+
+type DeviceInclusionSpec struct {
+	// A regular expression that will be used to include certain devices
+	// For example - "^rbd[0-9]+p?[0-9]{0,}$"
+	// +optional
+	DeviceNamePattern string `json:"deviceNameFilter"`
+
+	// The devices of this mechanicalPropery will be included.
+	// By default it is RotationalAndNonRotational(or SSD/HDD both)
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// The minimum size of the device which needs to be included
+	// +optional
+	MinSize *int `json:"minSize"`
+
+	// The maximum size of the device which needs to be included
+	// +optional
+	MaxSize *int `json:"maxSize"`
+
+	// The list of device models which need to be included
+	// +optional
+	Models []string `json:"model"`
+
+	// The list of device uuids which needs to be included
+	// +optional
+	UUID []string `json:"uuid"`
+
+	// Devices with these fsType will be included
+	// +optional
+	FSType []string `json:"fstype"`
+
+	// Devices with below labels will be included
+	// +optional
+	Labels []string `json:"label"`
+}
+```
+
+### Test Plan
+
+- The integration tests for the LSO already exist. These tests will need to be updated to test this feature.
+
+### Graduation Criteria
+
+N/A
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Removing a deprecated feature
+
+- None of the features are getting deprecated
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+N/A
+
+## Alternatives
```

Comment:
Please add more detail: Why should the proposed design be chosen over these alternatives?

Reply:
added